# generated by patch-package 6.4.10
#
# command:
#   npx patch-package ts-expose-internals
#
# declared package:
#   ts-expose-internals: ^4.8.4
#
diff --git a/node_modules/ts-expose-internals/typescript.d.ts b/node_modules/ts-expose-internals/typescript.d.ts
index 330d4ba..b9a6b92 100644
--- a/node_modules/ts-expose-internals/typescript.d.ts
+++ b/node_modules/ts-expose-internals/typescript.d.ts
@@ -12,7 +12,7 @@ MERCHANTABLITY OR NON-INFRINGEMENT.
 See the Apache Version 2.0 License for specific language governing permissions
 and limitations under the License.
 ***************************************************************************** */
-declare module "typescript" {
+declare module "typescript-full" {
     type GetIteratorCallback = <I extends readonly any[] | ReadonlySetShim<any> | ReadonlyMapShim<any, any> | undefined>(iterable: I) => IteratorShim<I extends ReadonlyMapShim<infer K, infer V> ? [
         K,
         V
@@ -71,7 +71,7 @@ declare module "typescript" {
     }
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     const versionMajorMinor = "4.8";
     /** The version of the TypeScript compiler release */
     const version: string;
@@ -173,7 +173,7 @@ declare module "typescript" {
     const Map: MapConstructor;
     const Set: SetConstructor;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function getIterator<I extends readonly any[] | ReadonlySet<any> | ReadonlyESMap<any, any> | undefined>(iterable: I): Iterator<I extends ReadonlyESMap<infer K, infer V> ? [
         K,
         V
@@ -796,7 +796,7 @@ declare module "typescript" {
      */
     const trimStringStart: (s: string) => string;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum LogLevel {
         Off = 0,
         Error = 1,
@@ -917,7 +917,7 @@ declare module "typescript" {
         function attachDebugPrototypeIfDebug(mapper: TypeMapper): TypeMapper;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Describes a precise semantic version number, https://semver.org
      */
@@ -946,7 +946,7 @@ declare module "typescript" {
         toString(): string;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface PerformanceHooks {
         /** Indicates whether we should write native performance events */
         shouldWriteNativeEvents: boolean;
@@ -982,7 +982,7 @@ declare module "typescript" {
     /** Gets a timestamp with (at least) ms resolution */
     const timestamp: () => number;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace performance {
         interface Timer {
             enter(): void;
@@ -1035,11 +1035,11 @@ declare module "typescript" {
         function disable(): void;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** Performance logger that will generate ETW events if possible - check for `logEvent` member, as `etwModule` will be `{}` when browserified */
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export let tracing: typeof tracingEnabled | undefined;
     namespace tracingEnabled {
         type Mode = "project" | "build" | "server";
@@ -1080,7 +1080,7 @@ declare module "typescript" {
     }
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export type Path = string & {
         __pathBrand: any;
     };
@@ -7549,7 +7549,7 @@ declare module "typescript" {
     }
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * djb2 hashing algorithm
      * http://www.cse.yorku.ca/~oz/hash.html
@@ -7798,7 +7798,7 @@ declare module "typescript" {
     export function setSys(s: System): void;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Internally, we represent paths as strings with '/' as the directory separator.
      * When we make system calls (eg: LanguageServiceHost.getDirectory()),
@@ -8185,7 +8185,7 @@ declare module "typescript" {
     function forEachAncestorDirectory<T>(directory: string, callback: (directory: string) => T | undefined): T | undefined;
     function isNodeModulesDirectory(dirPath: Path): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     const Diagnostics: {
         Unterminated_string_literal: DiagnosticMessage;
         Identifier_expected: DiagnosticMessage;
@@ -10047,7 +10047,7 @@ declare module "typescript" {
         _0_is_automatically_exported_here: DiagnosticMessage;
     };
 }
-declare module "typescript" {
+declare module "typescript-full" {
     type ErrorCallback = (message: DiagnosticMessage, length: number) => void;
     function tokenIsIdentifierOrKeyword(token: SyntaxKind): boolean;
     function tokenIsIdentifierOrKeywordOrGreaterThan(token: SyntaxKind): boolean;
@@ -10140,7 +10140,7 @@ declare module "typescript" {
     function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean, languageVariant?: LanguageVariant, textInitial?: string, onError?: ErrorCallback, start?: number, length?: number): Scanner;
     function utf16EncodeAsString(codePoint: number): string;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function isExternalModuleNameRelative(moduleName: string): boolean;
     function sortAndDeduplicateDiagnostics<T extends Diagnostic>(diagnostics: readonly T[]): SortedReadonlyArray<T>;
     function getDefaultLibFileName(options: CompilerOptions): string;
@@ -10539,7 +10539,7 @@ declare module "typescript" {
     function hasRestParameter(s: SignatureDeclaration | JSDocSignature): boolean;
     function isRestParameter(node: ParameterDeclaration | JSDocParameterTag): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export const resolvingEmptyArray: never[];
     export const externalHelpersModuleNameText = "tslib";
     export const defaultMaximumTruncationLength = 160;
@@ -11648,7 +11648,7 @@ declare module "typescript" {
     export function getParameterTypeNode(parameter: ParameterDeclaration | JSDocParameterTag): TypeNode | undefined;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * A `BaseNodeFactory` is an abstraction over an `ObjectAllocator` that handles caching `Node` constructors
      * and allocating `Node` instances based on a set of predefined types.
@@ -11665,15 +11665,15 @@ declare module "typescript" {
      */
     function createBaseNodeFactory(): BaseNodeFactory;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function createParenthesizerRules(factory: NodeFactory): ParenthesizerRules;
     const nullParenthesizerRules: ParenthesizerRules;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function createNodeConverters(factory: NodeFactory): NodeConverters;
     const nullNodeConverters: NodeConverters;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum NodeFactoryFlags {
         None = 0,
         NoParenthesizerRules = 1,
@@ -11705,7 +11705,7 @@ declare module "typescript" {
     function createSourceMapSource(fileName: string, text: string, skipTrivia?: (pos: number) => number): SourceMapSource;
     function setOriginalNode<T extends Node>(node: T, original: Node | undefined): T;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Associates a node with the current transformation, initializing
      * various transient transformation properties.
@@ -11809,7 +11809,7 @@ declare module "typescript" {
     function setTypeNode<T extends Node>(node: T, type: TypeNode): T;
     function getTypeNode<T extends Node>(node: T): TypeNode | undefined;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface EmitHelperFactory {
         getUnscopedHelperName(name: string): Identifier;
         createDecorateHelper(decoratorExpressions: readonly Expression[], target: Expression, memberName?: Expression, descriptor?: Expression): Expression;
@@ -11984,7 +11984,7 @@ declare module "typescript" {
     const advancedAsyncSuperHelper: EmitHelper;
     function isCallToHelper(firstSegment: Expression, helperName: __String): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function isNumericLiteral(node: Node): node is NumericLiteral;
     function isBigIntLiteral(node: Node): node is BigIntLiteral;
     function isStringLiteral(node: Node): node is StringLiteral;
@@ -12205,7 +12205,7 @@ declare module "typescript" {
     function isJSDocImplementsTag(node: Node): node is JSDocImplementsTag;
     function isSyntaxList(n: Node): n is SyntaxList;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function createEmptyExports(factory: NodeFactory): ExportDeclaration;
     function createMemberAccessForPropertyName(factory: NodeFactory, target: Expression, memberName: PropertyName, location?: TextRange): MemberExpression;
     function createJsxFactoryExpression(factory: NodeFactory, jsxFactoryEntity: EntityName | undefined, reactNamespace: string, parent: JsxOpeningLikeElement | JsxOpeningFragment): Expression;
@@ -12361,12 +12361,12 @@ declare module "typescript" {
     function elideNodes<T extends Node>(factory: NodeFactory, nodes: NodeArray<T>): NodeArray<T>;
     function elideNodes<T extends Node>(factory: NodeFactory, nodes: NodeArray<T> | undefined): NodeArray<T> | undefined;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function setTextRange<T extends TextRange>(range: T, location: TextRange | undefined): T;
     function canHaveModifiers(node: Node): node is HasModifiers;
     function canHaveDecorators(node: Node): node is HasDecorators;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * NOTE: You should not use this, it is only exported to support `createNode` in `~/src/deprecatedCompat/deprecations.ts`.
      */
@@ -12457,7 +12457,7 @@ declare module "typescript" {
     /** @internal */
     function tagNamesAreEquivalent(lhs: JsxTagNameExpression, rhs: JsxTagNameExpression): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export const compileOnSaveCommandLineOption: CommandLineOption;
     export const inverseJsxOptionMap: ESMap<string, string>;
     /**
@@ -12699,7 +12699,7 @@ declare module "typescript" {
     export function convertCompilerOptionsForTelemetry(opts: CompilerOptions): CompilerOptions;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export function trace(host: ModuleResolutionHost, message: DiagnosticMessage, ...args: any[]): void;
     export function isTraceEnabled(compilerOptions: CompilerOptions, host: ModuleResolutionHost): boolean;
     interface ModuleResolutionState {
@@ -12895,7 +12895,7 @@ declare module "typescript" {
     export function loadModuleFromGlobalCache(moduleName: string, projectName: string | undefined, compilerOptions: CompilerOptions, host: ModuleResolutionHost, globalCache: string, packageJsonInfoCache: PackageJsonInfoCache): ResolvedModuleWithFailedLookupLocations;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum ModuleInstanceState {
         NonInstantiated = 0,
         Instantiated = 1,
@@ -12905,10 +12905,10 @@ declare module "typescript" {
     function bindSourceFile(file: SourceFile, options: CompilerOptions): void;
     function isExportsOrModuleExportsOrAlias(sourceFile: SourceFile, node: Expression): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function createGetSymbolWalker(getRestTypeOfSignature: (sig: Signature) => Type, getTypePredicateOfSignature: (sig: Signature) => TypePredicate | undefined, getReturnTypeOfSignature: (sig: Signature) => Type, getBaseTypes: (type: Type) => Type[], resolveStructuredTypeMembers: (type: ObjectType) => ResolvedType, getTypeOfSymbol: (sym: Symbol) => Type, getResolvedSymbol: (node: Node) => Symbol, getConstraintOfTypeParameter: (typeParameter: TypeParameter) => Type | undefined, getFirstIdentifier: (node: EntityNameOrEntityNameExpression) => Identifier, getTypeArguments: (type: TypeReference) => readonly Type[]): (accept?: (symbol: Symbol) => boolean) => SymbolWalker;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum TypeFacts {
         None = 0,
         TypeofEQString = 1,
@@ -13011,7 +13011,7 @@ declare module "typescript" {
     function signatureHasRestParameter(s: Signature): boolean;
     function signatureHasLiteralTypes(s: Signature): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Visits a Node using the supplied visitor, possibly returning a new Node in its place.
      *
@@ -13103,7 +13103,7 @@ declare module "typescript" {
     function visitEachChild<T extends Node>(node: T | undefined, visitor: Visitor, context: TransformationContext, nodesVisitor?: typeof visitNodes, tokenVisitor?: Visitor): T | undefined;
     function visitEachChild<T extends Node>(node: T | undefined, visitor: Visitor, context: TransformationContext, nodesVisitor?: NodesVisitor, tokenVisitor?: Visitor, nodeVisitor?: NodeVisitor): T | undefined;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface SourceMapGeneratorOptions {
         extendedDiagnostics?: boolean;
     }
@@ -13143,7 +13143,7 @@ declare module "typescript" {
     function createDocumentPositionMapper(host: DocumentPositionMapperHost, map: RawSourceMap, mapPath: string): DocumentPositionMapper;
     const identitySourceMapConsumer: DocumentPositionMapper;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function getOriginalNodeId(node: Node): number;
     interface ExternalModuleInfo {
         externalImports: (ImportDeclaration | ImportEqualsDeclaration | ExportDeclaration)[];
@@ -13221,7 +13221,7 @@ declare module "typescript" {
      */
     function getAllDecoratorsOfClassElement(member: ClassElement, parent: ClassLikeDeclaration): AllDecorators | undefined;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum FlattenLevel {
         All = 0,
         ObjectRest = 1
@@ -13251,17 +13251,17 @@ declare module "typescript" {
      */
     function flattenDestructuringBinding(node: VariableDeclaration | ParameterDeclaration, visitor: (node: Node) => VisitResult<Node>, context: TransformationContext, level: FlattenLevel, rval?: Expression, hoistTempVariables?: boolean, skipInitializer?: boolean): VariableDeclaration[];
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum ProcessLevel {
         LiftRestriction = 0,
         All = 1
     }
     function processTaggedTemplateExpression(context: TransformationContext, node: TaggedTemplateExpression, visitor: Visitor, currentSourceFile: SourceFile, recordTaggedTemplateString: (temp: Identifier) => void, level: ProcessLevel): CallExpression | TaggedTemplateExpression;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformTypeScript(context: TransformationContext): (node: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum PrivateIdentifierKind {
         Field = "f",
         Method = "m",
@@ -13276,7 +13276,7 @@ declare module "typescript" {
      */
     function transformClassFields(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     type SerializedEntityName = Identifier | PropertyAccessEntityNameExpression;
     type SerializedTypeNode = SerializedEntityName | ConditionalExpression | VoidExpression;
     export interface RuntimeTypeSerializerContext {
@@ -13324,39 +13324,39 @@ declare module "typescript" {
     export function createRuntimeTypeSerializer(context: TransformationContext): RuntimeTypeSerializer;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformLegacyDecorators(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2017(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
     /** Creates a variable named `_super` with accessor properties for the given property names. */
     function createSuperAccessVariableStatement(factory: NodeFactory, resolver: EmitResolver, node: FunctionLikeDeclaration, names: Set<__String>): VariableStatement;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2018(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2019(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2020(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2021(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformESNext(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformJsx(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2016(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformES2015(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Transforms ES5 syntax into ES3 syntax.
      *
@@ -13364,22 +13364,22 @@ declare module "typescript" {
      */
     function transformES5(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformGenerators(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformModule(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformSystemModule(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformECMAScriptModule(context: TransformationContext): (x: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function transformNodeModule(context: TransformationContext): (node: SourceFile | Bundle) => SourceFile | Bundle;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     type GetSymbolAccessibilityDiagnostic = (symbolAccessibilityResult: SymbolAccessibilityResult) => (SymbolAccessibilityDiagnostic | undefined);
     interface SymbolAccessibilityDiagnostic {
         errorNode: Node;
@@ -13391,7 +13391,7 @@ declare module "typescript" {
     function createGetSymbolAccessibilityDiagnosticForNodeName(node: DeclarationDiagnosticProducing): (symbolAccessibilityResult: SymbolAccessibilityResult) => SymbolAccessibilityDiagnostic | undefined;
     function createGetSymbolAccessibilityDiagnosticForNode(node: DeclarationDiagnosticProducing): GetSymbolAccessibilityDiagnostic;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function getDeclarationDiagnostics(host: EmitHost, resolver: EmitResolver, file: SourceFile | undefined): DiagnosticWithLocation[] | undefined;
     function isInternalDeclaration(node: Node, currentSourceFile: SourceFile): boolean | 0 | undefined;
     /**
@@ -13406,7 +13406,7 @@ declare module "typescript" {
         (node: SourceFile | Bundle): SourceFile | Bundle;
     };
 }
-declare module "typescript" {
+declare module "typescript-full" {
     const noTransformers: EmitTransformers;
     function getTransformers(compilerOptions: CompilerOptions, customTransformers?: CustomTransformers, emitOnlyDtsFiles?: boolean): EmitTransformers;
     function noEmitSubstitution(_hint: EmitHint, node: Node): Node;
@@ -13424,7 +13424,7 @@ declare module "typescript" {
     function transformNodes<T extends Node>(resolver: EmitResolver | undefined, host: EmitHost | undefined, factory: NodeFactory, options: CompilerOptions, nodes: readonly T[], transformers: readonly TransformerFactory<T>[], allowDtsFiles: boolean): TransformationResult<T>;
     const nullTransformationContext: TransformationContext;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function isBuildInfoFile(file: string): boolean;
     /**
      * Iterates over the source files that are expected to have an emit output.
@@ -13463,7 +13463,7 @@ declare module "typescript" {
     function emitUsingBuildInfo(config: ParsedCommandLine, host: EmitUsingBuildInfoHost, getCommandLine: (ref: ProjectReference) => ParsedCommandLine | undefined, customTransformers?: CustomTransformers): EmitUsingBuildInfoResult;
     function createPrinter(printerOptions?: PrinterOptions, handlers?: PrintHandlers): Printer;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Partial interface of the System thats needed to support the caching of directory structure
      */
@@ -13573,7 +13573,7 @@ declare module "typescript" {
         watcher: FileWatcher;
     }>(objWithWatcher: T): void;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export function findConfigFile(searchPath: string, fileExists: (fileName: string) => boolean, configName?: string): string | undefined;
     export function resolveTripleslashReference(moduleName: string, containingFile: string): string;
     export function computeCommonSourceDirectoryOfFilenames(fileNames: readonly string[], currentDirectory: string, getCanonicalFileName: GetCanonicalFileName): string;
@@ -13753,7 +13753,7 @@ declare module "typescript" {
     export function getModuleNameStringLiteralAt({ imports, moduleAugmentations }: SourceFileImportsList, index: number): StringLiteralLike;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface EmitOutput {
         outputFiles: OutputFile[];
         emitSkipped: boolean;
@@ -13766,7 +13766,7 @@ declare module "typescript" {
         buildInfo?: BuildInfo;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function getFileEmitOutput(program: Program, sourceFile: SourceFile, emitOnlyDtsFiles: boolean, cancellationToken?: CancellationToken, customTransformers?: CustomTransformers, forceDtsEmit?: boolean): EmitOutput;
     interface BuilderState {
         /**
@@ -13878,7 +13878,7 @@ declare module "typescript" {
         function getAllFilesExcludingDefaultLibraryFile(state: BuilderState, programOfThisState: Program, firstSourceFile: SourceFile | undefined): readonly SourceFile[];
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface ReusableDiagnostic extends ReusableDiagnosticRelatedInformation {
         /** May store more in future. For now, this will simply be `true` to indicate when a diagnostic is an unused-identifier diagnostic. */
         reportsUnnecessary?: {};
@@ -14089,7 +14089,7 @@ declare module "typescript" {
         compilerOptions: CompilerOptions;
     }, configFileParsingDiagnostics: readonly Diagnostic[]): BuilderProgram;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     type AffectedFileResult<T> = {
         result: T;
         affected: SourceFile | Program;
@@ -14244,7 +14244,7 @@ declare module "typescript" {
     function createAbstractBuilder(newProgram: Program, host: BuilderProgramHost, oldProgram?: BuilderProgram, configFileParsingDiagnostics?: readonly Diagnostic[]): BuilderProgram;
     function createAbstractBuilder(rootNames: readonly string[] | undefined, options: CompilerOptions | undefined, host?: CompilerHost, oldProgram?: BuilderProgram, configFileParsingDiagnostics?: readonly Diagnostic[], projectReferences?: readonly ProjectReference[]): BuilderProgram;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** This is the cache of module/typedirectives resolution that can be retained across program */
     export interface ResolutionCache {
         startRecordingFilesWithChangedResolutions(): void;
@@ -14306,7 +14306,7 @@ declare module "typescript" {
     export function createResolutionCache(resolutionHost: ResolutionCacheHost, rootDirForResolution: string | undefined, logChangesWhenResolvingModule: boolean): ResolutionCache;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace moduleSpecifiers {
         function updateModuleSpecifier(compilerOptions: CompilerOptions, importingSourceFile: SourceFile, importingSourceFileName: Path, toFileName: string, host: ModuleSpecifierResolutionHost, oldImportSpecifier: string, options?: ModuleSpecifierOptions): string | undefined;
         function getModuleSpecifier(compilerOptions: CompilerOptions, importingSourceFile: SourceFile, importingSourceFileName: Path, toFileName: string, host: ModuleSpecifierResolutionHost, options?: ModuleSpecifierOptions): string;
@@ -14323,7 +14323,7 @@ declare module "typescript" {
         function tryGetJSExtensionForFile(fileName: string, options: CompilerOptions): Extension | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Create a function that reports error by writing to the system and handles the formatting of the diagnostic
      */
@@ -14441,7 +14441,7 @@ declare module "typescript" {
     export function performIncrementalCompilation(input: IncrementalCompilationOptions): ExitStatus.Success | ExitStatus.DiagnosticsPresent_OutputsSkipped | ExitStatus.DiagnosticsPresent_OutputsGenerated;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface ReadBuildProgramHost {
         useCaseSensitiveFileNames(): boolean;
         getCurrentDirectory(): string;
@@ -14602,7 +14602,7 @@ declare module "typescript" {
      */
     function createWatchProgram<T extends BuilderProgram>(host: WatchCompilerHostOfConfigFile<T>): WatchOfConfigFile<T>;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum UpToDateStatusType {
         Unbuildable = 0,
         UpToDate = 1,
@@ -14732,7 +14732,7 @@ declare module "typescript" {
     }
     function resolveConfigFileProjectName(project: string): ResolvedConfigFileName;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface BuildOptions {
         dry?: boolean;
         force?: boolean;
@@ -14857,7 +14857,7 @@ declare module "typescript" {
     }
     type InvalidatedProject<T extends BuilderProgram> = UpdateOutputFileStampsProject | BuildInvalidedProject<T> | UpdateBundleProject<T>;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace server {
         type ActionSet = "action::set";
         type ActionInvalidate = "action::invalidate";
@@ -14895,7 +14895,7 @@ declare module "typescript" {
         function nowString(): string;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace server {
         interface TypingInstallerResponse {
             readonly kind: ActionSet | ActionInvalidate | EventTypesRegistry | ActionPackageInstalled | EventBeginInstallTypes | EventEndInstallTypes | EventInitializationFailed;
@@ -14977,7 +14977,7 @@ declare module "typescript" {
         type TypingInstallerResponseUnion = SetTypings | InvalidateCachedTypings | TypesRegistryResponse | PackageInstalledResponse | InstallTypes | InitializationFailedResponse;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace JsTyping {
         interface TypingResolutionHost {
             directoryExists(path: string): boolean;
@@ -15035,7 +15035,7 @@ declare module "typescript" {
         function renderPackageNameValidationFailure(result: PackageNameValidationResult, typing: string): string;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface Node {
         getSourceFile(): SourceFile;
         getChildCount(sourceFile?: SourceFile): number;
@@ -16363,7 +16363,7 @@ interface PromiseConstructor {
     reject(reason: any): Promise<never>;
     all<T>(values: (T | PromiseLike<T>)[]): Promise<T[]>;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     const scanner: Scanner;
     enum SemanticMeaning {
         None = 0,
@@ -16741,7 +16741,7 @@ declare module "typescript" {
     function getFormatCodeSettingsForWriting({ options }: formatting.FormatContext, sourceFile: SourceFile): FormatCodeSettings;
     function jsxModeNeedsExplicitImport(jsx: JsxEmit | undefined): boolean;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum ImportKind {
         Named = 0,
         Default = 1,
@@ -16791,7 +16791,7 @@ declare module "typescript" {
         exportKind: ExportKind;
     } | undefined;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** The classifier is used for syntactic highlighting in editors via the TSServer */
     function createClassifier(): Classifier;
     function getSemanticClassifications(typeChecker: TypeChecker, cancellationToken: CancellationToken, sourceFile: SourceFile, classifiableNames: ReadonlySet<__String>, span: TextSpan): ClassifiedSpan[];
@@ -16799,7 +16799,7 @@ declare module "typescript" {
     function getSyntacticClassifications(cancellationToken: CancellationToken, sourceFile: SourceFile, span: TextSpan): ClassifiedSpan[];
     function getEncodedSyntacticClassifications(cancellationToken: CancellationToken, sourceFile: SourceFile, span: TextSpan): Classifications;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace classifier.v2020 {
         enum TokenEncodingConsts {
             typeOffset = 8,
@@ -16832,13 +16832,13 @@ declare module "typescript" {
         function getEncodedSemanticClassifications(program: Program, cancellationToken: CancellationToken, sourceFile: SourceFile, span: TextSpan): Classifications;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace Completions.StringCompletions {
         function getStringLiteralCompletions(sourceFile: SourceFile, position: number, contextToken: Node | undefined, options: CompilerOptions, host: LanguageServiceHost, program: Program, log: Log, preferences: UserPreferences): CompletionInfo | undefined;
         function getStringLiteralCompletionDetails(name: string, sourceFile: SourceFile, position: number, contextToken: Node | undefined, checker: TypeChecker, options: CompilerOptions, host: LanguageServiceHost, cancellationToken: CancellationToken, preferences: UserPreferences): CompletionEntryDetails | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace Completions {
         export const moduleSpecifierResolutionLimit = 100;
         export const moduleSpecifierResolutionCacheAttemptLimit = 1000;
@@ -16934,7 +16934,7 @@ declare module "typescript" {
         export {};
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface DocumentHighlights {
         fileName: string;
         highlightSpans: HighlightSpan[];
@@ -16943,7 +16943,7 @@ declare module "typescript" {
         function getDocumentHighlights(program: Program, cancellationToken: CancellationToken, sourceFile: SourceFile, position: number, sourceFilesToSearch: readonly SourceFile[]): DocumentHighlights[] | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * The document registry represents a store of SourceFile objects that can be shared between
      * multiple LanguageService instances. A LanguageService instance holds on the SourceFile (AST)
@@ -17044,7 +17044,7 @@ declare module "typescript" {
     };
     function createDocumentRegistryInternal(useCaseSensitiveFileNames?: boolean, currentDirectory?: string, externalCache?: ExternalDocumentCache): DocumentRegistry;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace FindAllReferences {
         interface ImportsResult {
             /** For every import of the symbol, the location and local symbol for the import. */
@@ -17107,7 +17107,7 @@ declare module "typescript" {
         function getExportInfo(exportSymbol: Symbol, exportKind: ExportKind, checker: TypeChecker): ExportInfo | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace FindAllReferences {
         interface SymbolAndEntries {
             readonly definition: Definition | undefined;
@@ -17239,7 +17239,7 @@ declare module "typescript" {
         }
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace CallHierarchy {
         type NamedExpression = ClassExpression & {
             name: Identifier;
@@ -17275,14 +17275,14 @@ declare module "typescript" {
         function getOutgoingCalls(program: Program, declaration: CallHierarchyDeclaration): CallHierarchyOutgoingCall[];
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     export function getEditsForFileRename(program: Program, oldFileOrDirPath: string, newFileOrDirPath: string, host: LanguageServiceHost, formatContext: formatting.FormatContext, preferences: UserPreferences, sourceMapper: SourceMapper): readonly FileTextChanges[];
     /** If 'path' refers to an old directory, returns path in the new directory. */
     type PathUpdater = (path: string) => string | undefined;
     export function getPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string, getCanonicalFileName: GetCanonicalFileName, sourceMapper: SourceMapper | undefined): PathUpdater;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace GoToDefinition {
         function getDefinitionAtPosition(program: Program, sourceFile: SourceFile, position: number, searchOtherFilesOnly?: boolean, stopAtAlias?: boolean): readonly DefinitionInfo[] | undefined;
         function getReferenceAtPosition(sourceFile: SourceFile, position: number, program: Program): {
@@ -17298,7 +17298,7 @@ declare module "typescript" {
         function findReferenceInPosition(refs: readonly FileReference[], pos: number): FileReference | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace JsDoc {
         function getJsDocCommentsFromDeclarations(declarations: readonly Declaration[], checker?: TypeChecker): SymbolDisplayPart[];
         function getJsDocTagsFromDeclarations(declarations?: Declaration[], checker?: TypeChecker): JSDocTagInfo[];
@@ -17334,18 +17334,18 @@ declare module "typescript" {
         function getDocCommentTemplateAtPosition(newLine: string, sourceFile: SourceFile, position: number, options?: DocCommentTemplateOptions): TextInsertion | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace NavigateTo {
         function getNavigateToItems(sourceFiles: readonly SourceFile[], checker: TypeChecker, cancellationToken: CancellationToken, searchValue: string, maxResultCount: number | undefined, excludeDtsFiles: boolean): NavigateToItem[];
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace NavigationBar {
         function getNavigationBarItems(sourceFile: SourceFile, cancellationToken: CancellationToken): NavigationBarItem[];
         function getNavigationTree(sourceFile: SourceFile, cancellationToken: CancellationToken): NavigationTree;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace OrganizeImports {
         /**
          * Organize imports by:
@@ -17371,12 +17371,12 @@ declare module "typescript" {
         function compareImportsOrRequireStatements(s1: AnyImportOrRequireStatement, s2: AnyImportOrRequireStatement): Comparison;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace OutliningElementsCollector {
         function collectElements(sourceFile: SourceFile, cancellationToken: CancellationToken): OutliningSpan[];
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     enum PatternMatchKind {
         exact = 0,
         prefix = 1,
@@ -17396,21 +17396,21 @@ declare module "typescript" {
     function breakIntoCharacterSpans(identifier: string): TextSpan[];
     function breakIntoWordSpans(identifier: string): TextSpan[];
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function preProcessFile(sourceText: string, readImportFiles?: boolean, detectJavaScriptImports?: boolean): PreProcessedFileInfo;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace Rename {
         function getRenameInfo(program: Program, sourceFile: SourceFile, position: number, preferences: UserPreferences): RenameInfo;
         function nodeIsEligibleForRename(node: Node): boolean;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace SmartSelectionRange {
         function getSmartSelectionRange(pos: number, sourceFile: SourceFile): SelectionRange;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace SignatureHelp {
         function getSignatureHelpItems(program: Program, sourceFile: SourceFile, position: number, triggerReason: SignatureHelpTriggerReason | undefined, cancellationToken: CancellationToken): SignatureHelpItems | undefined;
         interface ArgumentInfoForCompletions {
@@ -17421,12 +17421,12 @@ declare module "typescript" {
         function getArgumentInfoForCompletions(node: Node, position: number, sourceFile: SourceFile): ArgumentInfoForCompletions | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace InlayHints {
         function provideInlayHints(context: InlayHintsContext): InlayHint[];
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface SourceMapper {
         toLineColumnOffset(fileName: string, position: number): LineAndCharacter;
         tryGetSourcePosition(info: DocumentPosition): DocumentPosition | undefined;
@@ -17451,7 +17451,7 @@ declare module "typescript" {
     type ReadMapFile = (mapFileName: string, mapFileNameFromDts: string | undefined) => string | undefined | DocumentPositionMapper | false;
     function getDocumentPositionMapper(host: DocumentPositionMapperHost, generatedFileName: string, generatedFileLineInfo: LineInfo, readMapFile: ReadMapFile): DocumentPositionMapper | undefined;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     function computeSuggestionDiagnostics(sourceFile: SourceFile, program: Program, cancellationToken: CancellationToken): DiagnosticWithLocation[];
     function returnsPromise(node: FunctionLikeDeclaration, checker: TypeChecker): boolean;
     function isReturnStatementWithFixablePromiseHandler(node: Node, checker: TypeChecker): node is ReturnStatement & {
@@ -17460,7 +17460,7 @@ declare module "typescript" {
     function isFixablePromiseHandler(node: Node, checker: TypeChecker): boolean;
     function canBeConvertedToAsync(node: Node): node is FunctionDeclaration | MethodDeclaration | FunctionExpression | ArrowFunction;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace SymbolDisplay {
         function getSymbolKind(typeChecker: TypeChecker, symbol: Symbol, location: Node): ScriptElementKind;
         function getSymbolModifiers(typeChecker: TypeChecker, symbol: Symbol): string;
@@ -17473,7 +17473,7 @@ declare module "typescript" {
         function getSymbolDisplayPartsDocumentationAndSymbolKind(typeChecker: TypeChecker, symbol: Symbol, sourceFile: SourceFile, enclosingDeclaration: Node | undefined, location: Node, semanticMeaning?: SemanticMeaning, alias?: Symbol): SymbolDisplayPartsDocumentationAndSymbolKind;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface TranspileOptions {
         compilerOptions?: CompilerOptions;
         fileName?: string;
@@ -17492,7 +17492,7 @@ declare module "typescript" {
     /** JS users may pass in string values for enum compiler options (such as ModuleKind), so convert. */
     function fixupCompilerOptions(options: CompilerOptions, diagnostics: Diagnostic[]): CompilerOptions;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         enum FormattingRequestKind {
             FormatDocument = 0,
@@ -17528,7 +17528,7 @@ declare module "typescript" {
         }
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         interface FormattingScanner {
             advance(): void;
@@ -17545,7 +17545,7 @@ declare module "typescript" {
         function getFormattingScanner<T>(text: string, languageVariant: LanguageVariant, startPos: number, endPos: number, cb: (scanner: FormattingScanner) => T): T;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         interface Rule {
             readonly debugName: string;
@@ -17577,7 +17577,7 @@ declare module "typescript" {
         }
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         interface RuleSpec {
             readonly leftTokenRange: TokenRange;
@@ -17587,7 +17587,7 @@ declare module "typescript" {
         function getAllRules(): RuleSpec[];
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         function getFormatContext(options: FormatCodeSettings, host: FormattingHost): FormatContext;
         type RulesMap = (context: FormattingContext) => readonly Rule[] | undefined;
@@ -17601,7 +17601,7 @@ declare module "typescript" {
         }
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         interface FormatContext {
             readonly options: FormatCodeSettings;
@@ -17632,7 +17632,7 @@ declare module "typescript" {
         function getIndentationString(indentation: number, options: EditorSettings): string;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace formatting {
         namespace SmartIndenter {
             /**
@@ -17679,7 +17679,7 @@ declare module "typescript" {
         }
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace textChanges {
         export interface ConfigurableStart {
             leadingTriviaOption?: LeadingTriviaOption;
@@ -17867,7 +17867,7 @@ declare module "typescript" {
         export {};
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
         function createCodeFixActionWithoutFixAll(fixName: string, changes: FileTextChanges[], description: DiagnosticAndArguments): CodeFixAction;
         function createCodeFixAction(fixName: string, changes: FileTextChanges[], description: DiagnosticAndArguments, fixId: {}, fixAllDescription: DiagnosticAndArguments, command?: CodeActionCommand): CodeFixAction;
@@ -17882,7 +17882,7 @@ declare module "typescript" {
         function eachDiagnostic(context: CodeFixAllContext, errorCodes: readonly number[], cb: (diag: DiagnosticWithLocation) => void): void;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor {
         /** @param name An unique code associated with each refactor. Does not have to be human-readable. */
         function registerRefactor(name: string, refactor: Refactor): void;
@@ -17890,81 +17890,81 @@ declare module "typescript" {
         function getEditsForRefactor(context: RefactorContext, refactorName: string, actionName: string): RefactorEditInfo | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
         type DeclarationWithType = FunctionLikeDeclaration | VariableDeclaration | PropertySignature | PropertyDeclaration;
         function parameterShouldGetTypeFromJSDoc(node: Node): node is DeclarationWithType;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
         const importFixName = "import";
         /**
@@ -18002,135 +18002,135 @@ declare module "typescript" {
         function moduleSpecifierToValidIdentifier(moduleSpecifier: string, target: ScriptTarget | undefined, forceCapitalize?: boolean): string;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
         /**
          * Finds members of the resolved type that are missing in the class pointed to by class decl
@@ -18191,7 +18191,7 @@ declare module "typescript" {
         export {};
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
         type AcceptedDeclaration = ParameterPropertyDeclaration | PropertyDeclaration | PropertyAssignment;
         type AcceptedNameType = Identifier | StringLiteral;
@@ -18215,80 +18215,80 @@ declare module "typescript" {
         export {};
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace codefix {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor {
         function doChangeNamedToNamespaceOrDefault(sourceFile: SourceFile, program: Program, changes: textChanges.ChangeTracker, toConvert: NamedImports, shouldUseDefault?: boolean): void;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.convertToOptionalChainExpression {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.addOrRemoveBracesToArrowFunction {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.extractSymbol {
         /**
          * Compute the associated code actions
@@ -18372,15 +18372,15 @@ declare module "typescript" {
         }
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.generateGetAccessorAndSetAccessor {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor {
         /**
          * Returned by refactor functions when some error message needs to be surfaced to users.
@@ -18399,31 +18399,31 @@ declare module "typescript" {
         function refactorKindBeginsWith(known: string, requested: string | undefined): boolean;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.addOrRemoveBracesToArrowFunction {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.convertParamsToDestructuredObject {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.convertStringOrTemplateLiteral {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.convertArrowFunctionOrFunctionExpression {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace refactor.inferFunctionReturnType {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** The version of the language service API */
     const servicesVersion = "0.8";
     interface DisplayPartsSymbolWriter extends EmitTextWriter {
@@ -18465,7 +18465,7 @@ declare module "typescript" {
      */
     function getDefaultLibFilePath(options: CompilerOptions): string;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     namespace BreakpointResolver {
         /**
          * Get the breakpoint span in given sourceFile
@@ -18473,7 +18473,7 @@ declare module "typescript" {
         function spanInSourceFileAtLocation(sourceFile: SourceFile, position: number): TextSpan | undefined;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * Transform one or more nodes using the supplied transformers.
      * @param source A single `Node` or an array of `Node` objects.
@@ -18482,7 +18482,7 @@ declare module "typescript" {
      */
     function transform<T extends Node>(source: T | T[], transformers: TransformerFactory<T>[], compilerOptions?: CompilerOptions): TransformationResult<T>;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface ScriptSnapshotShim {
         /** Gets a portion of the script snapshot specified by [start, end). */
         getText(start: number, end: number): string;
@@ -18760,7 +18760,7 @@ declare module "typescript" {
         unregisterShim(shim: Shim): void;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** Defines a list of overloads by ordinal */
     type OverloadDefinitions = {
         readonly [P in number]: (...args: any[]) => any;
@@ -18798,7 +18798,7 @@ declare module "typescript" {
     export function buildOverload(name: string): OverloadBuilder;
     export {};
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** @deprecated Use `factory.createNodeArray` or the factory supplied by your transformation context instead. */
     const createNodeArray: <T extends Node>(elements?: readonly T[] | undefined, hasTrailingComma?: boolean | undefined) => NodeArray<T>;
     /** @deprecated Use `factory.createNumericLiteral` or the factory supplied by your transformation context instead. */
@@ -19658,11 +19658,11 @@ declare module "typescript" {
      */
     const getMutableClone: <T extends Node>(node: T) => T;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /** @deprecated Use `isTypeAssertionExpression` instead. */
     const isTypeAssertion: (node: Node) => node is TypeAssertion;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * @deprecated Use `ts.ReadonlyESMap<K, V>` instead.
      */
@@ -19674,13 +19674,13 @@ declare module "typescript" {
     interface Map<T> extends ESMap<string, T> {
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     /**
      * @deprecated Use `isMemberName` instead.
      */
     const isIdentifierOrPrivateIdentifier: (node: Node) => node is MemberName;
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface NodeFactory {
         /** @deprecated Use the overload that accepts 'modifiers' */
         createConstructorTypeNode(typeParameters: readonly TypeParameterDeclaration[] | undefined, parameters: readonly ParameterDeclaration[], type: TypeNode): ConstructorTypeNode;
@@ -19688,7 +19688,7 @@ declare module "typescript" {
         updateConstructorTypeNode(node: ConstructorTypeNode, typeParameters: NodeArray<TypeParameterDeclaration> | undefined, parameters: NodeArray<ParameterDeclaration>, type: TypeNode): ConstructorTypeNode;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface NodeFactory {
         createImportTypeNode(argument: TypeNode, assertions?: ImportTypeAssertionContainer, qualifier?: EntityName, typeArguments?: readonly TypeNode[], isTypeOf?: boolean): ImportTypeNode;
         /** @deprecated Use the overload that accepts 'assertions' */
@@ -19697,7 +19697,7 @@ declare module "typescript" {
         updateImportTypeNode(node: ImportTypeNode, argument: TypeNode, qualifier: EntityName | undefined, typeArguments: readonly TypeNode[] | undefined, isTypeOf?: boolean): ImportTypeNode;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface NodeFactory {
         /** @deprecated Use the overload that accepts 'modifiers' */
         createTypeParameterDeclaration(name: string | Identifier, constraint?: TypeNode, defaultType?: TypeNode): TypeParameterDeclaration;
@@ -19705,7 +19705,7 @@ declare module "typescript" {
         updateTypeParameterDeclaration(node: TypeParameterDeclaration, name: Identifier, constraint: TypeNode | undefined, defaultType: TypeNode | undefined): TypeParameterDeclaration;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
     interface Node {
         /**
          * @deprecated `decorators` has been removed from `Node` and merged with `modifiers` on the `Node` subtypes that support them.
@@ -19907,5 +19907,5 @@ declare module "typescript" {
         updateExportDeclaration(node: ExportDeclaration, decorators: readonly Decorator[] | undefined, modifiers: readonly Modifier[] | undefined, isTypeOnly: boolean, exportClause: NamedExportBindings | undefined, moduleSpecifier: Expression | undefined, assertClause: AssertClause | undefined): ExportDeclaration;
     }
 }
-declare module "typescript" {
+declare module "typescript-full" {
 }
